### PR TITLE
Tile Changer Area

### PR DESCRIPTION
**Pull request by ThatDan123**
-Click drag to fill areas

-Graffiti tiles

-Colour mode to set colour without changing tile type

-Fix bug with pressing ecs to close

### CODE REVIEW

File: UnityProject/Assets/Scripts/UI/Systems/AdminTools/DevTools/GUI_DevTileChanger.cs [hunk 1]
```diff
@@ -272,47 +295,70 @@ public void OnActionButtonClick(int buttonType)
...
+			if (CommonInput.GetMouseButtonUp(0) == false) return;
+
+			if(dragStartPos == MouseInputController.MouseWorldPosition.RoundToInt()) return;
+
+			//End drag
+			switch (currentAction)
+			{
+				case ActionType.Place:
+					//Also remove if shift is pressed when placing for quick remove
+					if (Input.GetKey(KeyCode.LeftShift) || Input.GetKey(KeyCode.RightShift))
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/UI/Systems/AdminTools/DevTools/GUI_DevTileChanger.cs:357]: Would it be sensible to do the inverse if in remove mode? Or do you see shift as to guarantee remove mode.

A little hint in the UI might be nice too if it doesn't have one already.

**Review by NoooneyDude — DISMISSED** [review]

**Comment by NoooneyDude** [comment]
Could consider sending a struct instead, might be a bit tidier => help shut up Codacy.

File: UnityProject/Assets/Scripts/UI/Systems/AdminTools/DevTools/GUI_DevTileChanger.cs (hunk 1 above)
Comment (AtnerNT) [reply UnityProject/Assets/Scripts/UI/Systems/AdminTools/DevTools/GUI_DevTileChanger.cs:357]: On the note of remove mode, I still find it funny how the remove tool for admins actually works correctly yet the unity editor version is still broke

**Comment by ThatDan123** [comment]
> Could consider sending a struct instead, might be a bit tidier => help shut up Codacy.

Will do that now

File: UnityProject/Assets/Scripts/UI/Systems/AdminTools/DevTools/GUI_DevTileChanger.cs (hunk 1 above)
Comment (ThatDan123) [reply UnityProject/Assets/Scripts/UI/Systems/AdminTools/DevTools/GUI_DevTileChanger.cs:357]: > Would it be sensible to do the inverse if in remove mode? Or do you see shift as to guarantee remove mode.
> 
> A little hint in the UI might be nice too if it doesn't have one already.

Have made it just do the reverse, not sure what UI thing i could add?

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/UI/Systems/AdminTools/DevTools/GUI_DevTileChanger.cs:357]: Could replace the mode buttons with w/ a toggle group and change the current mode string to a hint label. Up to you though.

**Comment by corp-0** [comment]
was it complaining because the methods had too many parameters? That is a fair concern, most of the time that is a symptom of a function just doing way too many things.

**Comment by NoooneyDude** [comment]
> was it complaining because the methods had too many parameters? That is a fair concern, most of the time that is a symptom of a function just doing way too many things.

In this case it was the entry point for client => server, so a struct is sensible.

**Review by NoooneyDude — APPROVED** [review]

**Pull request merged**